File: src_local/agents/Python/flow.py
from src_local.agents.base import BaseActivePersona
import logging
import time

logger = logging.getLogger(__name__)

class FlowPersona(BaseActivePersona):
    """Core: PhD in Control Theory 🌊"""
    
    def __init__(self, project_root):
        super().__init__(project_root)
        self.name, self.emoji, self.role, self.stack = "Flow", "🌊", "PhD Orchestration Architect", "Python"

    def perform_audit(self) -> list:
        start_time = time.time()
        logger.info(f"[{self.name}] Auditando Topologia de Fluxo...")
        
        audit_rules = [
            {'regex': r"if .*:\s+.*if .*:\s+.*if", 'issue': 'Complexidade: Arrow Pattern detectado.', 'severity': 'high'}
        ]
        
        results = self.find_patterns(('.py',), audit_rules)
        self._log_performance(start_time, len(results))
        return results

    def _reason_about_objective(self, objective, file, content):
        info = self.context_data.get(file, {})
        complexity = info.get("complexity", 1)
        if complexity > 15:
            return {
                "file": file,
                "issue": f"Entropia Lógica: Complexidade ({complexity}) excede Soberania (15)",
                "severity": "STRATEGIC",
                "context": self.name
            }
        return None

    def get_system_prompt(self):
        return f"Você é o Dr. {self.name}, mestre em fluxos."

File: src_local/agents/Bun/System/flow.ts
import { BaseActivePersona } from "../../base.ts";
import type { AuditFinding, AuditRule, StrategicFinding, ProjectContext } from "../../base.ts";

/**
 * 🔄 Dr. Flow — PhD in Bun Control Flow & Shell Scripting
 * Especialista em Bun.$ shell, Bun.spawn e fluxos de controle assíncronos.
 */
export class FlowPersona extends BaseActivePersona {
    constructor(projectRoot?: string) {
        super(projectRoot);
        this.name = "Flow";
        this.emoji = "🔄";
        this.role = "PhD Bun Control Flow Architect";
        this.phd_identity = "Bun Control Flow & Async Execution Strategies";
        this.stack = "Bun";
    }

    public override async execute(context: ProjectContext): Promise<(AuditFinding | StrategicFinding)[]> {
        this.setContext(context);
        const findings: (AuditFinding | StrategicFinding)[] = await this.performAudit();
        return findings;
    }

    override getAuditRules(): { extensions: string[]; rules: AuditRule[] } {
        return {
            extensions: ['.ts', '.tsx'],
            rules: [
                { regex: /\.then\([^)]*\)\.then\([^)]*\)\.then/, issue: 'Callback Hell: Cadeia .then() em Bun — use async/await PhD.', severity: 'high' },
                { regex: /Bun\.spawn\(\[[^\]]*\]\)(?![\s\S]{0,100}await)/, issue: 'Fire-and-Forget: Bun.spawn sem await — processo pode não completar PhD.', severity: 'high' },
                { regex: /Bun\.\$`[^`]*`(?![\s\S]{0,50}await|try)/, issue: 'Bun Shell: Bun.$ sem await ou try-catch — erro silenciado PhD.', severity: 'medium' },
            ]
        };
    }

    public override async performAudit(): Promise<AuditFinding[]> {
        this.startMetrics();
        const rules = this.getAuditRules();
        const results = await this.findPatterns(rules.extensions, rules.rules);
        this.endMetrics(results.length);
        return results;
    }

    public override reasonAboutObjective(objective: string, file: string, content: string | Promise<string | null>): StrategicFinding | null {
        if (typeof content === 'string' && /\.then\([^)]*\)\.then\([^)]*\)\.then/.test(content)) {
            return {
                file, severity: "HIGH",
                issue: `Entropia: O objetivo '${objective}' exige clareza. Em '${file}', callback chains obscurecem o fluxo Bun PhD.`,
                context: "Callback hell detected"
            };
        }
        return {
            file, severity: "INFO",
            issue: `PhD Flow (Bun): Analisando fluxos ciclomáticos de concorrência para ${objective}.`,
            context: this.name
        };
    }

    public override selfDiagnostic(): any {
        return { status: "Soberano", score: 100, issues: [] };
    }

    public override getSystemPrompt(): string {
        return `Você é o Dr. ${this.name}, mestre em fluxos de controle e shell scripting Bun.`;
    }
}
